api/V1/recordingUtil.js

```diff
@@ -246,85 +246,62 @@ async function get(request, type) {
     throw new ClientError("No file found with given datapoint.");
   }
 
-  const downloadFileData = {
-    _type: "fileDownload",
-    key: recording.fileKey,
-    filename: recording.getFileName(),
-    mimeType: recording.fileMimeType
+  const data = {
+    recording: handleLegacyTagFieldsForGetOnRecording(recording)
   };
 
-  const downloadRawData = {
-    _type: "fileDownload",
-    key: recording.rawFileKey,
-    filename: recording.getRawFileName(),
-    mimeType: recording.rawMimeType
-  };
-
-  let rawSize = null;
-  if (recording.rawFileKey) {
-    await util
-      .getS3Object(recording.rawFileKey)
-      .then(rawS3Data => {
-        rawSize = rawS3Data.ContentLength;
-      })
-      .catch(err => {
-        log.warn(
-          "Error retrieving S3 Object for recording",
-          err.message,
-          recording.rawFileKey
-        );
-      });
-  }
-
-  let cookedSize = null;
   if (recording.fileKey) {
-    await util
-      .getS3Object(recording.fileKey)
-      .then(s3Data => {
-        cookedSize = s3Data.ContentLength;
-      })
-      .catch(err => {
-        log.warn(
-          "Error retrieving S3 Object for recording",
-          err.message,
-          recording.fileKey
-        );
-      });
+    data.cookedJWT = jsonwebtoken.sign({
+      _type: "fileDownload",
+      key: recording.fileKey,
+      filename: recording.getFileName(),
+      mimeType: recording.fileMimeType
+    },
+    config.server.passportSecret,
+    { expiresIn: 60 * 10 }
+    );
+    data.cookedSize = await util.getS3ObjectFileSize(recording.fileKey);
   }
 
-  delete recording.rawFileKey;
```

Comment:
I've removed this line, as it's still being returned currently because the `handleLegacyTagFieldsForGetOnRecording` method recreates the object with that key back in, so this deletion is currently pointless.

Do we actually want/care about this being deleted, and if so are there other keys that should also be deleted?

Reply:
Good catch! 

We don't really want `rawFileKey` and `fileKey` to leak out of the API as they are internal details only. If it's easy, could you arrange things so they don't leak out here, or file an issue if it's going to be a big change?

Reply:
I've re-added the delete down the bottom where it'll now have the correct effect. Still feels a little prone to leakage as it relies on those delete statements at the end, but it's back to as it was at least.